### PR TITLE
Working with payloads

### PR DESCRIPTION
**Pull request by nicolasparrague**

### CODE REVIEW

**Comment by SantiagoParraGuerrero** [comment]
@nicolasparrague you are failing a job

File: src-lookup-sample/main/default/classes/AccountContactLookup.cls [hunk 1]
```diff
@@ -0,0 +1,103 @@
+public class AccountContactLookup implements LookupResult.ILookupResult {
+    public final static Map<String, Object> ACCOUNT_ICON = new Map<String, Object>{
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/classes/AccountContactLookup.cls:2]: Why is this called account icon?

File: src-lookup-sample/main/default/classes/AccountLookup.cls [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        for (Account account : [SELECT Id, Name FROM Account WHERE Id IN: selectedIds]) {
+            
+            result.add(buildResult(account));
+        }
+    
+        return result;
+    }
+
+    public List<LookupResult> getSearchResults(
+    String searchTerm,
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/classes/AccountLookup.cls:19]: Can you use a code format?  You keep making code formatting mistakes

File: src-lookup-sample/main/default/classes/AccountLookup.cls [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    return result;
+    }
+
+    public List<LookupResult> getDefaultSearchResults(Map<String, Object> payload) {
+        List<LookupResult> result = new List<LookupResult>();
+
+        for (Account account : [SELECT Id, Name FROM Account LIMIT 10]) {
+            result.add(buildResult(account));
+            System.debug(account);
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/classes/AccountLookup.cls:45]: Remove debug statements

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            <c-lookup
+              data-id="accountLookup"
+              lwc:ref="lookup"
+              unique-id="AccountLookup" 
+              label="Account Search"
+              onchange={handleAccountChange}
+              value={accountValue}
+              required
+            ></c-lookup>
+              <p>{accountText}</p>
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.html:20]: Do

Your account selected was {selectedAccount.Name}

Do the same for the contact

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  @track contactPayload;
+  accountValue;
+  contactValue;
+  _accountText;
+  _contactText;
+
+  @api
+  get accountText() {
+    return this._accountText;
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js:14]: Why a get set for this?
Does not me make sense to me

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  set accountText(value) {
+    this._accountText = value;
+  }
+
+  @api
+  get contactText() {
+    return this._contactText;
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js:23]: Same here 
Why is this marked as api?

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  handleAccountChange(event) {
+
+    this.contactValue = [];
+    this._accountText = "";
+    this._contactText = "";
+    this.accountValue = event.detail.value[0];
+    this.contactPayload = { accountId : event.detail.value[0] };
+
+    if (this.accountValue !== undefined) {
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js:38]: Do 
If !this.account
Instead

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+      this._accountText = `Your account selected was ${account.name}`;
+    }
+  }
+
+  handleContactChange(event) {
+
+    this._contactText = "";
+    this.contactValue = event.detail.value[0];
+
+    if (this.accountValue === undefined || this.contactValue !== undefined) {
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js:49]: Same here

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+      const contact = event.detail.payload[this.contactValue];
+      this._contactText = `Your contact selected was ${contact.name} and its phone number is  ${contact.phone}`;
+    }
+  }
+
+  get accountValueEmpty() {
+    return !this.accountValue;
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js:56]: Don't need this
Just use a lwc:if this.account on your html

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js-meta.xml [hunk 10]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<LightningComponentBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>59.0</apiVersion>
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js-meta.xml:3]: Make sure you are using the latest api version checkout other files

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.html [hunk 11]
```diff
@@ -0,0 +1,38 @@
+<template>
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.html:1]: Why is this file called lookup account it is just a lookup account it does more than than

Called it 

workingWithPayloads instead

File: src-lookup-sample/main/default/lwc/lookupAccount/__tests__/lookupAccount.test.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+        // Act
+        const accountLookup = element.shadowRoot.querySelector('c-lookup[data-id="accountLookup"]');
+        accountLookup.dispatchEvent(accountChangeEvent);
+
+        const contactLookup = element.shadowRoot.querySelector('c-lookup[data-id="contactLookup"]');
+        contactLookup.dispatchEvent(contactChangeEvent);
+
+        // Assert
+        expect(element.accountText).toBe('Your account selected was Account name');
```
Comment (SantiagoParraGuerrero) [inline src-lookup-sample/main/default/lwc/lookupAccount/__tests__/lookupAccount.test.js:46]: You can't make an api variable just for the purpose of exposing it to the test class that is not Right

**Review by SantiagoParraGuerrero — CHANGES_REQUESTED** [review]

File: src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js (hunk 9 above)
Comment (nicolasparrague) [reply src-lookup-sample/main/default/lwc/lookupAccount/lookupAccount.js:56]: the requirement says that the contact must be deactivated when you do not select the account

**Review by SantiagoParraGuerrero — APPROVED** [review]

**Pull request merged**